### DISCUSSION TITLE
Removal of histogram stretch options in viewer

### DISCUSSION THREAD

**Discussion by robintw** [developmentseed/titiler#419] [Q&A]
I've noticed that in the latest version of titiler, the options for histogram stretching (min-max, 2% etc) in the viewer at `{URL}/cog/viewer` seem to have been removed. Is there a particular reason for this removal? And are there any plans to reintroduce this functionality?

I found these options very useful as part of demoing the abilities of titiler with COGs within my organisation, allowing me to show that the visualisation can be changed 'on the fly' and the visualisation can be quickly updated. I know I could implement the same sort of features in my own webmap interface using titiler, but it was nice to have it included with titiler.

**Comment by vincentsarago** [comment]
@robintw sorry for this. I removed it because I tried to simplify the viewer (and I'm not a frontend dev). I could try to add it back if I find some time!

**Reply by robintw** [reply]
Ah ok, thank you - that makes sense.

I'm also not a front-end developer, but I might be able to put together a PR to add it back in. Is it ok if I create an issue to add it back in, and I'll see if I can try and tackle it, and if not then someone else might be able to.

**Reply by vincentsarago** [reply]
Sure. 
It's not an easy task IMO. 

right now we allow stretching between min/max value (which default to datatype min/max) with UI defined in https://github.com/developmentseed/titiler/blob/master/src/titiler/application/titiler/application/templates/cog_index.html#L252-L257

If the datatype in of type `Byte` we don't allow this (for STAC I think it's enabled by default). we could make histogram stretching available by default (but optional) and then allow either fixed bounds or percentile based.

**Reply by robintw** [reply]
Ok, I'll create an issue and look into it.
